shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/route/future/engine/dml/ShadowUpdateStatementRoutingEngine.java

```diff
@@ -17,20 +17,89 @@
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 
 /**
  * Shadow update statement routing engine.
  */
-public final class ShadowUpdateStatementRoutingEngine implements ShadowRouteEngine {
+@RequiredArgsConstructor
+public final class ShadowUpdateStatementRoutingEngine extends AbstractShadowRouteEngine {
+    
+    private final UpdateStatementContext updateStatementContext;
+    
+    private final List<Object> parameters;
+    
+    @Override
+    protected Optional<Map<String, Collection<Comparable<?>>>> parseColumnValuesMappings() {
+        Map<String, Collection<Comparable<?>>> result = new LinkedHashMap<>();
+        Optional<WhereSegment> where = updateStatementContext.getWhere();
+        where.ifPresent(whereSegment -> parseWhereSegment(whereSegment, result));
+        return result.isEmpty() ? Optional.empty() : Optional.of(result);
+    }
+    
+    private void parseWhereSegment(final WhereSegment whereSegment, final Map<String, Collection<Comparable<?>>> columnValuesMappings) {
+        ExpressionExtractUtil.getAndPredicates(whereSegment.getExpr()).forEach(each -> parseAndPredicate(each.getPredicates(), columnValuesMappings));
+    }
+    
+    private void parseAndPredicate(final Collection<ExpressionSegment> predicates, final Map<String, Collection<Comparable<?>>> columnValuesMappings) {
+        for (ExpressionSegment each : predicates) {
+            parseExpressionSegment(each, columnValuesMappings);
+        }
+    }
+    
+    private void parseExpressionSegment(final ExpressionSegment expressionSegment, final Map<String, Collection<Comparable<?>>> columnValuesMappings) {
+        if (expressionSegment instanceof BinaryOperationExpression) {
+            parseBinaryOperationExpression((BinaryOperationExpression) expressionSegment, columnValuesMappings);
+        }
+        if (expressionSegment instanceof InExpression) {
+            parseInExpression((InExpression) expressionSegment, columnValuesMappings);
+        }
+    }
+    
+    private void parseInExpression(final InExpression expression, final Map<String, Collection<Comparable<?>>> columnValuesMappings) {
+        Optional<String> columnName = ShadowExtractor.extractColumnName(expression);
+        columnName.ifPresent(s -> ShadowExtractor.extractValues(expression.getRight(), parameters).ifPresent(values -> columnValuesMappings.put(s, values)));
+    }
+    
+    private void parseBinaryOperationExpression(final BinaryOperationExpression expression, final Map<String, Collection<Comparable<?>>> columnValuesMappings) {
+        Optional<String> columnName = ShadowExtractor.extractColumnName(expression);
+        columnName.ifPresent(s -> ShadowExtractor.extractValues(expression.getRight(), parameters).ifPresent(values -> columnValuesMappings.put(s, values)));
+    }
+    
+    @Override
+    protected ShadowDetermineCondition createShadowDetermineCondition() {
```

Comment:
@soulasuna Why not consider extracting a ShadowConditionEngine to handle these logics like ShardingConditionEngine?